File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        
        List<Integer> divide = new ArrayList<>();
        
        for (int i : arr) {
            
            if (i % divisor == 0) {
                divide.add(i);
            } else {
                continue;
            }
        }
        
        if (divide.size() > 0) {
            
            Collections.sort(divide);
        
            return divide.stream().mapToInt(Integer::intValue).toArray();
        } else {
            int[] a = new int[]{-1};
            return a;
        }
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.kt
class Solution {
	fun solution(arr: IntArray, divisor: Int): IntArray {
		val filter = arr.filter { it % divisor == 0 }.toMutableList()

		return if (filter.isNotEmpty()) {
			filter.sort()
			filter.toIntArray()
		} else {
			filter.add(-1)
			filter.toIntArray()
		}
	}
}
